crates/bevy_reflect/src/impls/std.rs

```diff
@@ -112,6 +112,20 @@ impl<T: Reflect> Reflect for Vec<T> {
     }
 }
 
+impl<T: Reflect + Clone> GetTypeRegistration for Vec<T> {
+    fn get_type_registration() -> TypeRegistration {
+        let mut registration = TypeRegistration::of::<Vec<T>>();
+        registration.insert::<Vec<T>>(FromType::<Vec<T>>::from_type());
+        registration
+    }
+}
```

Comment:
The "insert" line comes from `#reflect(Deserialize)`. It's not needed for `Serialize`, `Hash` and `PartialEq`, only for others (`Deserialize`, `Component`, `MapEntity` and any other reflected trait).
```suggestion
impl<T: Reflect + Clone> GetTypeRegistration for Vec<T> {
    fn get_type_registration() -> TypeRegistration {
        let mut registration = TypeRegistration::of::<Vec<T>>();
        registration.insert::<ReflectDeserialize>(FromType::<Vec<T>>::from_type());
        registration
    }
}
```

Reply:
Thank you! That line was the one I understood the least.

Does that mean we should determine a set of applicable traits that Vec<T> reflects and include a `registration.insert` for each of them?